apps/heft/src/plugins/PackageJsonConfigurationPlugin.ts

```diff
@@ -0,0 +1,34 @@
+import { IHeftPlugin } from '../pluginFramework/IHeftPlugin';
+import { HeftSession } from '../pluginFramework/HeftSession';
+import { HeftConfiguration } from '../configuration/HeftConfiguration';
+import { ITypescriptConfiguration, IBuildActionContext, ICompileStage } from '../cli/actions/BuildAction';
+import { IPackageJson } from '@rushstack/node-core-library';
+
+const PLUGIN_NAME: string = 'PackageJsonConfigurationPlugin';
+
+export class PackageJsonConfigurationPlugin implements IHeftPlugin {
+  public readonly displayName: string = PLUGIN_NAME;
+
+  public apply(heftSession: HeftSession, heftConfiguration: HeftConfiguration): void {
+    heftSession.hooks.build.tap(PLUGIN_NAME, (build: IBuildActionContext) => {
+      build.hooks.compile.tap(PLUGIN_NAME, (compile: ICompileStage) => {
+        compile.hooks.afterConfigureTypescript.tap(PLUGIN_NAME, () => {
+          this._updateTypescriptConfiguration(
+            heftConfiguration.projectPackageJson,
+            compile.properties.typescriptConfiguration
+          );
+        });
+      });
+    });
+  }
+
+  private _updateTypescriptConfiguration(
+    projectPackageJson: IPackageJson,
+    typescriptConfiguration: ITypescriptConfiguration
+  ): void {
+    if (projectPackageJson.private !== true) {
+      // Copy if the package is intended to be published
+      typescriptConfiguration.copyFromCacheMode = 'copy';
```

Comment:
It is somewhat counterintuitive that the `"private": true` field in **package.json** would influence the compiler. (It's often not even accurate in a Rush repo.)

Suggestions:
1. When this heuristic is applied, write something to the logs like 
   ```
   Defaulting to copyFromCacheMode="copy" because private=true in package.json
   ``` 
2. Make this a default that can be overridden easily via a simple config file (without making a rig)